pkg/driver/bacnet/merge/poll.go

```diff
@@ -38,18 +37,19 @@ func startPoll(init context.Context, name string, pollDelay, pollTimeout time.Du
 // Returns early with error if
 //
 //  1. ctx is done
-//  2. the number of polls is tries
-//  3. pollPeer returns an error
+//  2. pollPeer returns an error
+//  3. defaultDelay has passed if the ctx doesn't have a non-zero ctx.Deadline assigned
+//     or time.Now() + defaultDelay is earlier than the deadline assigned to ctx
 //
 // An backoff delay will be added between each call to pollPeer
-func pollUntil[T any](ctx context.Context, tries int, poll func(context.Context) (T, error), test func(T) bool) (T, error) {
+func pollUntil[T any](ctx context.Context, defaultDelay time.Duration, poll func(context.Context) (T, error), test func(T) bool) (T, error) {
```

Comment:
Sorry to be picky, but `delay` reads like things are going to pause for a bit. Can't this just be called `timeout` or something like that?